include/pybind11/detail/common.h

```diff
@@ -263,6 +263,19 @@ extern "C" {
     }                                                                          \
     PyObject *pybind11_init()
 
+// Returns nullptr on an error.
+#define GET_PYBIND11_MODULE(pybind11_init_, name, module) \
+  try {                                                   \
+    PYBIND11_CONCAT(pybind11_init_, name)(module);        \
+    return module.ptr();                                  \
+  } catch (pybind11::error_already_set & e) {             \
```

Comment:
If we're replacing `PYBIND11_CATCH_INIT_EXCEPTIONS` with this function, let's do it consistently and remove the macro completely, which means replacing the macro in the deprecated `PYBIND11_PLUGIN` as well.

Reply:
That was not intentional. Merged the latest changes which should make it more clear.